src/metadata.jl

```diff
@@ -34,12 +39,14 @@ function _pkgs()
 end
 
 function _env()
-    if isempty(_CACHE[:PROJECT]) && isempty(_CACHE[:MANIFEST])
-        _CACHE[:PROJECT] = read(Base.active_project(), String)
-        _CACHE[:MANIFEST] = read(
-            joinpath(dirname(Base.active_project()), "Manifest.toml"),
-            String
-        )
+    lock(_CACHE_LOCK) do
+        if isempty(_CACHE[:PROJECT]) && isempty(_CACHE[:MANIFEST])
+            _CACHE[:PROJECT] = read(Base.active_project(), String)
+            _CACHE[:MANIFEST] = read(
+                joinpath(dirname(Base.active_project()), "Manifest.toml"),
+                String
+            )
+        end
     end
 
     return (_CACHE[:PROJECT], _CACHE[:MANIFEST])
```

Comment:
I would feel more confident that this was correct if this line was within the lock.
Since reading a Dict while another thread is writing it is not thread safe.

I think this might be safe because the cache is write-once.
But let's make it obviously safe and save having to think.

Reply:
Done.